pkg/controller/util/federated_informer.go

```diff
@@ -324,6 +325,8 @@ func (f *federatedInformerImpl) GetClientForCluster(clusterName string) (generic
 	f.Lock()
 	defer f.Unlock()
 
+	clientConnectionStart := time.Now()
```

Comment:
Should we start timing before the lock is acquired?

Reply:
yes